[easy] fix nested view call taking in more than one -1

mikaylagawarecki opened the pull request:
https://github.com/pytorch/pytorch/pull/85691 (allowing only one -1 in nested view/reshape) broke this. Was not caught by CI but internal tests are broken

Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #86134

mikaylagawarecki committed:
[easy] fix nested view call taking in more than one -1

[ghstack-poisoned]

mikaylagawarecki commented on aten/src/ATen/native/transformers/transformer.cpp:
```diff
@@ -118,7 +118,7 @@ Tensor transformer_encoder_layer_forward(
      x = x.view({x.size(0), -1, 3, num_heads, embed_dim / num_heads});
      x = flash_attention_helper(x, x, x, 0.0, false);
-     x = x.view({-1, -1, embed_dim});
+     x = x.view({{x.size(0) * 3, -1, embed_dim});
```
is this the correct shape

mikaylagawarecki committed:
Update on "[easy] fix nested view call taking in more than one -1"

https://github.com/pytorch/pytorch/pull/85691 (allowing only one -1 in nested view/reshape) broke this. Was not caught by CI but internal tests are broken




[ghstack-poisoned]

cpuhrsch commented on aten/src/ATen/native/transformers/transformer.cpp:
```diff
@@ -116,9 +116,10 @@ Tensor transformer_encoder_layer_forward(
+     x = x.view({x_size_0, -1, 3, num_heads, embed_dim / num_heads});
      x = flash_attention_helper(x, x, x, 0.0, false);
-     x = x.view({-1, -1, embed_dim});
+     x = x.view({{x_size_0 * 3, -1, embed_dim});
```
I don't think this is right. flash_attention shouldn't change the batch size.

cpuhrsch reviewed (changes requested):
See comment

mikaylagawarecki committed:
Update on "[easy] fix nested view call taking in more than one -1"

https://github.com/pytorch/pytorch/pull/85691 (allowing only one -1 in nested view/reshape) broke this. Was not caught by CI but internal tests are broken




[ghstack-poisoned]

mikaylagawarecki commented on aten/src/ATen/native/transformers/transformer.cpp:
```diff
@@ -116,9 +116,10 @@ Tensor transformer_encoder_layer_forward(
+     x = x.view({x_size_0, -1, 3, num_heads, embed_dim / num_heads});
      x = flash_attention_helper(x, x, x, 0.0, false);
-     x = x.view({-1, -1, embed_dim});
+     x = x.view({{x_size_0, -1, embed_dim});
```
@cpuhrsch just to confirm this should not `x_size_0 * 3` here is that correct?

mikaylagawarecki commented:
@pytorchbot merge
